src/ivoc/ocmatrix.cpp

```diff
@@ -230,9 +230,9 @@ void OcFullMatrix::setdiag(int k, Vect* in) {
     } else {
         for (i = -k, j = 0; i < row && j < col; ++i, ++j) {
 #ifdef WIN32
-            m_set_val(m_, i, j, v_elem(in, i));
+            m_set_val(m_, i, j, v_elem(in, j));
```

Comment:
The semantics of `in` is slightly different, see:
https://nrn.readthedocs.io/en/8.2.2/python/programming/math/matrix.html#Matrix.setdiag

Reply:
fixed